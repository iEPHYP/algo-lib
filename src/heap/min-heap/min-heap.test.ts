import { MinHeap } from './min-heap';

describe('MinHeap', () => {
  let maxHeap = new MinHeap();

  beforeEach(() => {
    maxHeap = new MinHeap();
  });

  it('inserts items properly', () => {
    maxHeap.insert(1);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(5);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(3);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(4);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(2);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(0);
    expect(maxHeap.toList()).toMatchSnapshot();
    maxHeap.insert(10);
    expect(maxHeap.toList()).toMatchSnapshot();
  });

  it('extracts Min root properly', () => {
    maxHeap.insert(1);
    expect(maxHeap.extractMin()).toBe(1);
    expect(maxHeap.toList()).toMatchSnapshot('extracting from heap with 1 item');
    maxHeap.insert(1);
    maxHeap.insert(5);
    expect(maxHeap.extractMin()).toBe(1);
    expect(maxHeap.toList()).toMatchSnapshot('extracting from heap with 2 items');
    maxHeap.insert(1);
    maxHeap.insert(0);
    maxHeap.insert(4);
    maxHeap.insert(2);
    maxHeap.insert(3);
    maxHeap.insert(10);
    maxHeap.insert(6);
    maxHeap.insert(19);
    expect(maxHeap.toList()).toMatchSnapshot('heap with several items');
    expect(maxHeap.extractMin()).toBe(0);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 0 from heap with several items');
    expect(maxHeap.extractMin()).toBe(1);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 1 from heap with several items');
    expect(maxHeap.extractMin()).toBe(2);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 2 from heap with several items');
    expect(maxHeap.extractMin()).toBe(3);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 3 from heap with several items');
    expect(maxHeap.extractMin()).toBe(4);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 4 from heap with several items');
    expect(maxHeap.extractMin()).toBe(5);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 5 from heap with several items');
    expect(maxHeap.extractMin()).toBe(6);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 6 from heap with several items');
    expect(maxHeap.extractMin()).toBe(10);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 10 from heap with several items');
    expect(maxHeap.extractMin()).toBe(19);
    expect(maxHeap.toList()).toMatchSnapshot('extracting 19 from heap with several items');
  });
});
