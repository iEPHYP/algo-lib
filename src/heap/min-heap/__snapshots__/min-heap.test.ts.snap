// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinHeap extracts Max root properly: extracting 0 from heap with several items 1`] = `
[
  1,
  2,
  3,
  5,
  4,
  19,
  10,
  6,
]
`;

exports[`MinHeap extracts Max root properly: extracting 1 from heap with several items 1`] = `
[
  2,
  4,
  3,
  5,
  6,
  19,
  10,
]
`;

exports[`MinHeap extracts Max root properly: extracting 2 from heap with several items 1`] = `
[
  3,
  4,
  10,
  5,
  6,
  19,
]
`;

exports[`MinHeap extracts Max root properly: extracting 3 from heap with several items 1`] = `
[
  4,
  5,
  10,
  19,
  6,
]
`;

exports[`MinHeap extracts Max root properly: extracting 4 from heap with several items 1`] = `
[
  5,
  6,
  10,
  19,
]
`;

exports[`MinHeap extracts Max root properly: extracting 5 from heap with several items 1`] = `
[
  6,
  19,
  10,
]
`;

exports[`MinHeap extracts Max root properly: extracting 6 from heap with several items 1`] = `
[
  10,
  19,
]
`;

exports[`MinHeap extracts Max root properly: extracting 10 from heap with several items 1`] = `
[
  19,
]
`;

exports[`MinHeap extracts Max root properly: extracting 19 from heap with several items 1`] = `[]`;

exports[`MinHeap extracts Max root properly: extracting from heap with 1 item 1`] = `[]`;

exports[`MinHeap extracts Max root properly: extracting from heap with 2 items 1`] = `
[
  5,
]
`;

exports[`MinHeap extracts Max root properly: heap with several items 1`] = `
[
  0,
  2,
  1,
  5,
  4,
  3,
  10,
  6,
  19,
]
`;

exports[`MinHeap extracts Min root properly: extracting 0 from heap with several items 1`] = `
[
  1,
  2,
  3,
  5,
  4,
  19,
  10,
  6,
]
`;

exports[`MinHeap extracts Min root properly: extracting 1 from heap with several items 1`] = `
[
  2,
  4,
  3,
  5,
  6,
  19,
  10,
]
`;

exports[`MinHeap extracts Min root properly: extracting 2 from heap with several items 1`] = `
[
  3,
  4,
  10,
  5,
  6,
  19,
]
`;

exports[`MinHeap extracts Min root properly: extracting 3 from heap with several items 1`] = `
[
  4,
  5,
  10,
  19,
  6,
]
`;

exports[`MinHeap extracts Min root properly: extracting 4 from heap with several items 1`] = `
[
  5,
  6,
  10,
  19,
]
`;

exports[`MinHeap extracts Min root properly: extracting 5 from heap with several items 1`] = `
[
  6,
  19,
  10,
]
`;

exports[`MinHeap extracts Min root properly: extracting 6 from heap with several items 1`] = `
[
  10,
  19,
]
`;

exports[`MinHeap extracts Min root properly: extracting 10 from heap with several items 1`] = `
[
  19,
]
`;

exports[`MinHeap extracts Min root properly: extracting 19 from heap with several items 1`] = `[]`;

exports[`MinHeap extracts Min root properly: extracting from heap with 1 item 1`] = `[]`;

exports[`MinHeap extracts Min root properly: extracting from heap with 2 items 1`] = `
[
  5,
]
`;

exports[`MinHeap extracts Min root properly: heap with several items 1`] = `
[
  0,
  2,
  1,
  5,
  4,
  3,
  10,
  6,
  19,
]
`;

exports[`MinHeap inserts items properly 1`] = `
[
  1,
]
`;

exports[`MinHeap inserts items properly 2`] = `
[
  1,
  5,
]
`;

exports[`MinHeap inserts items properly 3`] = `
[
  1,
  5,
  3,
]
`;

exports[`MinHeap inserts items properly 4`] = `
[
  1,
  4,
  3,
  5,
]
`;

exports[`MinHeap inserts items properly 5`] = `
[
  1,
  2,
  3,
  5,
  4,
]
`;

exports[`MinHeap inserts items properly 6`] = `
[
  0,
  2,
  1,
  5,
  4,
  3,
]
`;

exports[`MinHeap inserts items properly 7`] = `
[
  0,
  2,
  1,
  5,
  4,
  3,
  10,
]
`;
